language/bert/onnxruntime_SUT.py

```diff
@@ -55,26 +56,45 @@ def __init__(self, args):
 
     def issue_queries(self, query_samples):
         for i in range(len(query_samples)):
+
             eval_features = self.qsl.get_features(query_samples[i].index)
-            if self.quantized:
-                fd = {
-                    "input_ids": np.array(eval_features.input_ids).astype(np.int64)[np.newaxis, :],
-                    "attention_mask": np.array(eval_features.input_mask).astype(np.int64)[np.newaxis, :],
-                    "token_type_ids": np.array(eval_features.segment_ids).astype(np.int64)[np.newaxis, :]
-                }
-            else:
-                fd = {
-                    "input_ids": np.array(eval_features.input_ids).astype(np.int64)[np.newaxis, :],
-                    "input_mask": np.array(eval_features.input_mask).astype(np.int64)[np.newaxis, :],
-                    "segment_ids": np.array(eval_features.segment_ids).astype(np.int64)[np.newaxis, :]
-                }
-            scores = self.sess.run([o.name for o in self.sess.get_outputs()], fd)
-            output = np.stack(scores, axis=-1)[0]
-
-            response_array = array.array("B", output.tobytes())
-            bi = response_array.buffer_info()
-            response = lg.QuerySampleResponse(query_samples[i].id, bi[0], bi[1])
-            lg.QuerySamplesComplete([response])
+            self.process_sample(eval_features, query_samples[i].id)
+
+    def process_sample(self, eval_features, query_id):
+
+        '''For Loadgen over the network'''
+        if self.network == "sut":
+            input_ids = eval_features['input_ids']
+            input_mask = eval_features['input_mask']
+            segment_ids = eval_features['segment_ids']
+        else:
+            input_ids = eval_features.input_ids
+            input_mask = eval_features.input_mask
+            segment_ids = eval_features.segment_ids
+
+        if self.quantized:
```

Comment:
Maybe better to quantize in QSL?

Reply:
In the reference implementations, quantization is applicable only for onnxruntime and the qsl implementation is shared by all the backends. Probably that's why quantization is kept in the onnxruntime backend code. This code is shared by the network SUT and also the non-network onnxruntime backend.